src/GitVersionCore/BuildServers/TeamCity.cs

```diff
@@ -4,9 +4,11 @@
 
     public class TeamCity : BuildServerBase
     {
-        public override bool CanApplyToCurrentContext()
+		public const string EnvironmentVariableName = "TEAMCITY_VERSION";
```

Comment:
Minor issue: Wouldn't a static readonly variable fit better? For cases where this value is used in another assembly and the core assembly is replaced (But most probably this won't happen anyway :)